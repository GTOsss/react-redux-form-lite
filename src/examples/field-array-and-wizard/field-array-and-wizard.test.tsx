import ReduxThunkTester from 'redux-thunk-tester';
import {applyMiddleware, combineReducers, createStore} from 'redux';
import {reducer} from '../../index';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import getFieldArrayAndWizard from './field-array-and-wizard';
import React from 'react';

const renderComponent = (onSubmit?) => {
  const reduxThunkTester = new ReduxThunkTester();
  const FieldArrayAndWizard = getFieldArrayAndWizard();

  const store = createStore(
    combineReducers({reduxForm: reducer}),
    applyMiddleware(reduxThunkTester.createReduxThunkHistoryMiddleware()),
  );

  const component = mount(
    <Provider store={store}><FieldArrayAndWizard onSubmit={onSubmit} /></Provider>,
  );

  return {
    reduxThunkTester,
    getStep1: (): IReduxFormState<{}> => store.getState().reduxForm.step1 as IReduxFormState<{}>,
    getStep2: (): IReduxFormState<{}> => store.getState().reduxForm.step2 as IReduxFormState<{}>,
    getWizardExample: (): IReduxFormWizard<{}> =>
      store.getState().reduxForm.fieldArrayWizard as IReduxFormWizard<{}>,
    component,
  };
};

describe('<FieldArrayAndWizard />', () => {
  test('Render field-array form (inputs).', () => {
    const {component} = renderComponent();
    expect(component.find('input')).toMatchSnapshot();
  });

  test('Render field-array form (store Step1)', () => {
    const {getStep1} = renderComponent();
    expect(getStep1()).toMatchSnapshot();
  });

  test('Store field-array submit Step1 (store)', () => {
    const {component, getStep1} = renderComponent();
    component.find({name: 'step1Input'}).at(0)
      .simulate('change', {target: {value: 'test first name 0'}});
    component.find('form').simulate('submit');
    expect(getStep1()).toMatchSnapshot();
  });

  test('Store field-array submit Step1 (inputs)', () => {
    const {component} = renderComponent();
    component.find({name: 'step1Input'}).at(0)
      .simulate('change', {target: {value: 'test first name 0'}});
    component.find('form').simulate('submit');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('Render field-array after push user (store Step2)', () => {
    const {component, getStep2} = renderComponent();
    component.find({name: 'step1Input'}).at(0)
      .simulate('change', {target: {value: 'test first name 0'}});
    component.find('form').simulate('submit');
    component.find('#addUser').simulate('click');
    component.find('#addUser').simulate('click');
    component.find({name: 'users[0].firstName'}).at(0)
      .simulate('change', {target: {value: 'test first name 0'}});
    component.find({name: 'users[0].secondName'}).at(0)
      .simulate('change', {target: {value: 'test second name 0'}});
    component.find({name: 'users[0].about.hobby'}).at(0)
      .simulate('change', {target: {value: 'test about hobby 0'}});
    component.find({name: 'users[1].firstName'}).at(1)
      .simulate('change', {target: {value: 'test first name 1'}});
    component.find({name: 'users[1].secondName'}).at(1)
      .simulate('change', {target: {value: 'test second name 1'}});
    component.find({name: 'users[1].about.hobby'}).at(1)
      .simulate('change', {target: {value: 'test about hobby 1'}});
    component.find('form').simulate('submit');
    expect(getStep2()).toMatchSnapshot();
  });

  test('Render field-array after push user (store wizard)', () => {
    const {component, getWizardExample} = renderComponent();
    component.find({name: 'step1Input'}).at(0)
      .simulate('change', {target: {value: 'test first name 0'}});
    component.find('form').simulate('submit');
    component.find('#addUser').simulate('click');
    component.find('#addUser').simulate('click');
    component.find({name: 'users[0].firstName'}).at(0)
      .simulate('change', {target: {value: 'test first name 0'}});
    component.find({name: 'users[0].secondName'}).at(0)
      .simulate('change', {target: {value: 'test second name 0'}});
    component.find({name: 'users[0].about.hobby'}).at(0)
      .simulate('change', {target: {value: 'test about hobby 0'}});
    component.find({name: 'users[1].firstName'}).at(1)
      .simulate('change', {target: {value: 'test first name 1'}});
    component.find({name: 'users[1].secondName'}).at(1)
      .simulate('change', {target: {value: 'test second name 1'}});
    component.find('form').simulate('submit');
    expect(getWizardExample()).toMatchSnapshot();
  });
});
