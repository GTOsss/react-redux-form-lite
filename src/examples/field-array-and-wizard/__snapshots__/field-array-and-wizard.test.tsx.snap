// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldArrayAndWizard /> Render field-array after push user (store Step2) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {
      "groupName": "Field required.",
    },
    "focused": false,
    "hasErrors": true,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "groupName": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "Field required.",
      "focused": false,
      "warning": "",
    },
    "users": Array [
      Object {
        "about": Object {
          "hobby": Object {
            "active": false,
            "blurred": false,
            "changed": true,
            "error": "",
            "focused": false,
            "warning": "",
          },
        },
        "firstName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
        "secondName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
      Object {
        "about": Object {
          "hobby": Object {
            "active": false,
            "blurred": false,
            "changed": true,
            "error": "",
            "focused": false,
            "warning": "",
          },
        },
        "firstName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
        "secondName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
    ],
  },
  "values": Object {
    "groupName": undefined,
    "users": Array [
      Object {
        "about": Object {
          "hobby": "test about hobby 0",
        },
        "firstName": "test first name 0",
        "friends": Array [],
        "id": 0,
        "secondName": "test second name 0",
      },
      Object {
        "about": Object {
          "hobby": "test about hobby 1",
        },
        "firstName": "test first name 1",
        "friends": Array [],
        "id": 1,
        "secondName": "test second name 1",
      },
    ],
  },
}
`;

exports[`<FieldArrayAndWizard /> Render field-array after push user (store wizard) 1`] = `
Object {
  "values": Object {
    "groupName": undefined,
    "step1Input": "test first name 0",
    "users": Array [
      Object {
        "about": Object {
          "hobby": "test about hobby 0",
        },
        "firstName": "test first name 0",
        "friends": Array [],
        "secondName": "test second name 0",
      },
      Object {
        "about": Object {
          "hobby": undefined,
        },
        "firstName": "test first name 1",
        "friends": Array [],
        "secondName": "test second name 1",
      },
    ],
  },
  "wizard": Object {
    "errorsMap": Object {
      "groupName": "Field required.",
      "users": Array [
        undefined,
        Object {
          "about": Object {
            "hobby": "Field required.",
          },
        },
      ],
    },
    "hasErrors": true,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<FieldArrayAndWizard /> Render field-array form (inputs). 1`] = `
<input
  name="step1Input"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  type="text"
  value=""
/>
`;

exports[`<FieldArrayAndWizard /> Render field-array form (store Step1) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "step1Input": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "",
      "focused": false,
      "warning": "",
    },
  },
  "values": Object {
    "step1Input": undefined,
  },
}
`;

exports[`<FieldArrayAndWizard /> Store field-array submit Step1 (inputs) 1`] = `
<input
  name="groupName"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  type="text"
  value=""
/>
`;

exports[`<FieldArrayAndWizard /> Store field-array submit Step1 (store) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "step1Input": Object {
      "active": false,
      "blurred": false,
      "changed": true,
      "error": "",
      "focused": false,
      "warning": "",
    },
  },
  "values": Object {
    "step1Input": "test first name 0",
  },
}
`;
